enhancement/KAD-4015_pro_mosaic_gallery

gilbert-hernandez opened the pull request:
[https://stellarwp.atlassian.net/browse/KAD-4015](https://stellarwp.atlassian.net/browse/KAD-4015)

This adds a mosiac gallery for the pro plugin.

mark-c-woodard reviewed (changes requested):
Nice feature!

Box shadows and captions background color transparency (when "below image, show always") were looking wrong to me.

gilbert-hernandez commented:
I addressed the issues mentioned above. The gap and grid row size are now responsive. Also, the box shadows have been removed from the mosaic gallery. Also, the captions below the image are now showing correctly on the front end.

mark-c-woodard commented:
@gilbert-hernandez Captions below the image with mosaic were working for me if I had a link active, but not when the link settings are set to "none"
